src/channel/anonymouscheck.js

```diff
@@ -0,0 +1,26 @@
+var ChannelModule = require("./module");
+var Flags = require("../flags");
+
+function AnonymousCheck(channel) {
+    ChannelModule.apply(this, arguments);
+}
+
+AnonymousCheck.prototype = Object.create(ChannelModule.prototype);
+
+AnonymousCheck.prototype.onUserPreJoin = function (user, data, cb) {
+    var chan = this.channel,
+    opts = this.channel.modules.options;
+   var anonymousBanned =  opts.get("block_anonymous_users");
+   if(anonymousBanned && user.isAnonymous()) {
+       user.socket.emit("errorMsg", { msg : "This channel has blocked anonymous users. Please provide a user name to join."});
```

Comment:
If you click OK without thinking about it, this could get confusing because it says both "Not Connected" (on the userlist) and "Connected" in the chat, and in reality the user is half-connected (connected to the server, but not the channel, yet).

The password feature actually has this same issue, but in that case if you enter the wrong password, it immediately opens another prompt.  Maybe doing something similar here would be a short term solution until the UI is reworked for these cases.

Reply:
Well, if we would want to re-display the message anytime they clicked OK, I think instead of opening another prompt we might just want to remove the ability for the user to press O.K.  / dismiss the prompt.  How does that sound?  Alternatively, if there's a 'better' way to display a more persistent message in the cancel in addition to the dialog, that seems like it would clear up any confusion.

Reply:
What I meant was you could replace the OK prompt with a similar one to the one that the password feature uses, which requires you to enter a name to continue.

Reply:
Right on. That's totally something I could rig up. The only concern I would have is do you think it might a weird experience for users to have two inputs for guest login? That is, the one they're used to in the chat area and the dialog prompting them.

Reply:
It is a little weird, but I think the whole experience is already weird.  Ideally both this and the channel password case should block the whole UI until the user actually joins the channel (but I think that's a separate task).

Reply:
Ok, uh, do you think you could provide some acceptance criteria for what the ideal behaviour would be for this and channel password? I'd be down to take on the task if it's something that you think is worth adding.

Reply:
I don't have a mockup or anything, but what I was imagining was that when the client received the socket frame asking for a password or guest login, it would cover the whole channel UI with a full page modal (except the navbar) and center the prompt to enter the required information.  I'm open to discussion on this, I just think that showing the entire UI as if it is connected, while the client is only half connected, is confusing.

Reply:
Yeah, I totally agree with it being confusing with regards to the connected message in the channel area

Um, so I can't think of anything that seems bad about your suggestion here, which if I understand correctly is covering the channel UI to hide the connected message and providing the user with an input to enter whatever relevant data.  

The only alternative I can think of is keeping the behavior the way it is for the password / anonymous challenge, but instead of displaying the connected message in chat, displaying a message like 'Connection Pending' in the cases where the attempt to join the channel is prevented by a password or anonymous login. 

I'm not married to this alternative, and I'm not sure how uh, feasible it would be to implement off hand, but it was first alternate route that comes to mind.  Unless it seems better to you, or you have other ideas, I think your proposed approach is pretty solid, and that's something I would be able to implement - reviewing pending of course.